linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -339,3 +339,142 @@
+        insertafter: "^submenu.*English.*{#"
+        line: '  set default="autoinstall"'
+
+    - name: "Update boot parameters for autoinstall"
+      ansible.builtin.blockinfile:
+        path: "{{ src_iso_file_dir }}/grub.cfg"
+        insertafter: ".*autoinstall.*"
+        block: |
+          # menuentry for unattended installation
+            menuentry "Automated installation" --id autoinstall {
+              set background_color=black
+              linux    /install/gtk/vmlinuz auto=true file=/cdrom/{{ unattend_install_file_name }} video=vesa:ywrap,mtrr vga=788 --- quiet
+              initrd   /install/gtk/initrd.gz
+            }
+
+    - name: "Update md5sum for Pardus ISO files"
+      ansible.builtin.shell: "{{ item }}"
+      with_items:
+        - "sed -i '#./isolinux/isolinux.cfg#d' md5sum.txt"
+        - "echo \"`md5sum isolinux.cfg | awk '{print $1}'` ./isolinux/isolinux.cfg\" >>md5sum.txt"
+        - "sed -i '#./boot/grub/grub.cfg#d' md5sum.txt"
+        - "echo \"`md5sum grub.cfg | awk '{print $1}'` ./boot/grub/grub.cfg\" >>md5sum.txt"
+        - "sed -i '#./isolinux/install_en.cfg#d' md5sum.txt"
+        - "echo \"`md5sum install_en.cfg | awk '{print $1}'` ./isolinux/install_en.cfg\" >>md5sum.txt"
+        - "sed -i '#./isolinux/install.cfg#d' md5sum.txt"
+        - "echo \"`md5sum install.cfg | awk '{print $1}'` ./isolinux/install.cfg\" >>md5sum.txt"
+      args:
+        chdir: "{{ src_iso_file_dir }}"
+      register: update_initrd_output
+
+    - name: "Print command output for updating initrd"
+      ansible.builtin.debug: var=update_initrd_output
+    
+    - name: "Get file to get code name"
+      ansible.builtin.shell: "cat {{ src_iso_file_dir }}/md5sum.txt | grep -E 'dists\\/[a-zA-Z]+\\/Release$' | cut -d. -f 2"
+      register: release_file_path
+    
+    - name: "Print release file path"
+      ansible.builtin.debug: var=release_file_path
+    
+    - name: "Extract specific release file inside ISO"
+      community.general.iso_extract:
+        image: "{{ src_iso_file_path }}"
+        dest: "{{ src_iso_file_dir }}"
+        files:
+          - "{{ release_file_path.stdout.split('/', 1)[-1] }}"
+    
+    - name: "Get codename of the ISO distribution"
+      ansible.builtin.shell: "cat {{ src_iso_file_dir }}/Release | grep Codename | cut -d: -f 2"
+      register: codename_output
+    
+    - name: "Set facts of the codename"
+      ansible.builtin.set_fact:
+        os_codename: "{{ codename_output.stdout.strip() }}"
+
+    - name: "Print codename"
+      ansible.builtin.debug: var=os_codename
+
+    - name: "Replace the codename in file preseed.cfg"
+      ansible.builtin.replace:
+        path: "{{ new_unattend_install_conf }}"
+        regexp: "yirmibir"
+        replace: "{{ os_codename }}"
+    
+    - name: "Print the content of unattend install config file after replace codename"
+      ansible.builtin.debug:
+        msg: "{{ lookup('file', new_unattend_install_conf) | split('\n') }}"
```

Comment:
If you get codename in preseed late-command, you can remove line 428~461

Reply:
I need to replace the codename in preseed.cfg before customizing the ISO (putting the file preseed.cfg to orginal ISO) in order to this file preseed.cfg will work for future pardus distribution.

Thanks
Yuhua Zou